CUJExecutor/app/src/androidTest/java/com/example/firstapp/UserCujTest.java

```diff
@@ -30,31 +30,59 @@
 
 public class UserCujTest {
 
-    private static long TIMEOUTMS = 15000;
+    private static long TIMEOUTMS = 30000;
 
     @Test
     public void userIterateAndMeasureCuj() throws Exception {
         Bundle extras = InstrumentationRegistry.getArguments();
         String preCUJ = extras.getString("pre");
         String postCUJ = extras.getString("post");
         int iterations = Integer.parseInt(extras.getString("iters"));
-        boolean recordIntent = "r".equals(extras.getString("rec"));
+        String recStr = extras.getString("rec");
+        boolean recordIntent = ("r".equals(recStr) || "f".equals(recStr)); //r -> we want median clip, f -> we want full recording only
+        long recstart =  Long.parseLong(extras.getString("recstart"));
 
         ShellUtility shellUtility = new ShellUtility(TIMEOUTMS);
-        shellUtility.iterateAndMeasureCuj(preCUJ, postCUJ, iterations, recordIntent);
+        shellUtility.iterateAndMeasureCuj(preCUJ, postCUJ, iterations, recordIntent, recstart);
     }
 
     @Test
     public void userWalkCujNTimes() throws Exception {
+        ShellUtility shellUtility = new ShellUtility(TIMEOUTMS);
 
         Bundle extras = InstrumentationRegistry.getArguments();
         String preCUJ = extras.getString("pre");
         String postCUJ = extras.getString("post");
-        String include = extras.getString("include");
-        int n = Integer.parseInt(extras.getString("n"));
-        boolean includeMeasured = "c".equals(include); //tells us whether we should execute the the entire CUJ or just prepatory actions
+        String sectionFlag = extras.getString("include"); //Tells us which section of the CUJ to walk through
+        int n = Integer.parseInt(extras.getString("n")); // number of iterations
 
-        ShellUtility shellUtility = new ShellUtility(TIMEOUTMS);
-        shellUtility.walkCujNTimes(preCUJ, postCUJ, includeMeasured, n);
+        ShellUtility.cujFlag flag;
+        switch (sectionFlag) {
+            case "w" :
```

Comment:
Should use an Enum to represent the flags.  And consider using words to present the flags.

Reply:
Hmm not sure what you mean by this. Here, I'm converting the Strings passed from the script to an enum class (ShellUtility.cujFlag). Do you mean that the values passed from the script should be Enums?

Reply:
Ooops, you're right.  By the way, `cujFlag` should be capitalized.